operator/controller/src/test/java/io/apicurio/registry/operator/it/SmokeITTest.java

```diff
@@ -37,37 +39,45 @@ void smoke() {
-                    .getReadyReplicas()).isEqualTo(1);
-            assertThat(client.apps().deployments().inNamespace(namespace)
-                    .withName(registry.getMetadata().getName() + "-ui-deployment").get().getStatus()
-                    .getReadyReplicas()).isEqualTo(1);
-            return true;
-        });
+        checkDeploymentExists(registry, COMPONENT_APP, 1);
+        checkDeploymentExists(registry, COMPONENT_UI, 1);
 
         // Services
-        await().ignoreExceptions().until(() -> {
-            assertThat(client.services().inNamespace(namespace)
-                    .withName(registry.getMetadata().getName() + "-app-service").get().getSpec()
-                    .getClusterIP()).isNotBlank();
-            assertThat(client.services().inNamespace(namespace)
-                    .withName(registry.getMetadata().getName() + "-ui-service").get().getSpec()
-                    .getClusterIP()).isNotBlank();
-            return true;
-        });
+        checkServiceExists(registry, COMPONENT_APP);
+        checkServiceExists(registry, COMPONENT_UI);
 
         // Ingresses
-        await().ignoreExceptions().until(() -> {
-            assertThat(client.network().v1().ingresses().inNamespace(namespace)
-                    .withName(registry.getMetadata().getName() + "-app-ingress").get().getSpec().getRules()
-                    .get(0).getHost()).isEqualTo(registry.getSpec().getApp().getHost());
-            assertThat(client.network().v1().ingresses().inNamespace(namespace)
-                    .withName(registry.getMetadata().getName() + "-ui-ingress").get().getSpec().getRules()
-                    .get(0).getHost()).isEqualTo(registry.getSpec().getUi().getHost());
-            return true;
-        });
+        checkIngressExists(registry, COMPONENT_APP);
+        checkIngressExists(registry, COMPONENT_UI);
+    }
+
+    @Test
+    void replicas() {
+
+        var registry = ResourceFactory.deserialize("/k8s/examples/simple.apicurioregistry3.yaml",
+                ApicurioRegistry3.class);
+
+        registry.getMetadata().setNamespace(namespace);
+        registry.getSpec().getApp().setHost(ingressManager.getIngressHost("app"));
+        registry.getSpec().getUi().setHost(ingressManager.getIngressHost("ui"));
+
+        client.resource(registry).create();
+
+        // Verify first replica
+        checkDeploymentExists(registry, COMPONENT_APP, 1);
+        checkDeploymentExists(registry, COMPONENT_UI, 1);
+
+        // Scale up
+        registry.getSpec().getApp().setReplicas(3);
+        registry.getSpec().getUi().setReplicas(3);
+        checkDeploymentExists(registry, COMPONENT_APP, 3);
+        checkDeploymentExists(registry, COMPONENT_UI, 3);
+
+        // Scale down
+        registry.getSpec().getApp().setReplicas(2);
+        registry.getSpec().getUi().setReplicas(2);
+        checkDeploymentExists(registry, COMPONENT_APP, 2);
+        checkDeploymentExists(registry, COMPONENT_UI, 2);
```

Comment:
Don't we also need a check to ensure that extra replicas do *not* exist?  e.g. after line 80 something like:

```
checkDeploymentDoesNotExist(registry, COMPONENT_APP, 3);
```

Reply:
The check inside the code checks that the number of replicas ready is equal to the number passed, so it's already happening. Or are you suggesting checking for *all* replicas, no matter if they're ready or no?

Reply:
Oh no I misunderstood the parameter - I didn't look closely enough at it.  My bad.